server/lib/currency.ts

```diff
@@ -269,3 +270,32 @@ export function reduceArrayToCurrency(array: AmountWithCurrencyAndDate[], curren
     },
   );
 }
+
+/**
+ * Gets exchange rate between given crypto currency and fiat.
+ */
+export async function getExchangeRate(fromCurrency: string, toCurrency: string): Promise<string> {
+  try {
+    if (SUPPORTED_CRYPTO_CURRENCIES.includes(fromCurrency) && !SUPPORTED_CRYPTO_CURRENCIES.includes(toCurrency)) {
+      const { rate: cryptoToUSDRate } = await getCryptoToUSDRate(fromCurrency);
+      const USDtoCurrencyRate = await getFxRate('USD', toCurrency);
+      return cryptoToUSDRate * USDtoCurrencyRate;
+    } else if (
+      SUPPORTED_CRYPTO_CURRENCIES.includes(toCurrency) &&
+      !SUPPORTED_CRYPTO_CURRENCIES.includes(fromCurrency)
+    ) {
+      const { rate: cryptoToUSDRate } = await getCryptoToUSDRate(toCurrency);
+      const fromCurrencyUSDRate = await getFxRate(fromCurrency, 'USD');
+      return fromCurrencyUSDRate / cryptoToUSDRate;
+    } else if (SUPPORTED_CRYPTO_CURRENCIES.includes(fromCurrency) && SUPPORTED_CRYPTO_CURRENCIES.includes(toCurrency)) {
+      const cryptoFromCurrencyToUSD = await getCryptoToUSDRate(fromCurrency);
+      const cryptoToCurrencyToUSD = await getCryptoToUSDRate(toCurrency);
+      return cryptoFromCurrencyToUSD / cryptoToCurrencyToUSD;
+    } else {
+      return getFxRate(fromCurrency, toCurrency);
```

Comment:
Let's move the generic case and have it first.

Reply:
Done. 👍🏽